Description: Make applications launched from the update icon properly themed.
Bug-Ubuntu: https://bugs.launchpad.net/707916
Author: Felix Geyer <debfx-pkg@fobos.de>

--- kpackagekit-0.6.3.3.orig/SmartIcon/KpkUpdateIcon.cpp
+++ kpackagekit-0.6.3.3/SmartIcon/KpkUpdateIcon.cpp
@@ -31,6 +31,7 @@
 #include <KConfigGroup>
 #include <KActionCollection>
 #include <KMenu>
+#include <KToolInvocation>
 
 #include <KDebug>
 
@@ -53,7 +54,7 @@ KpkUpdateIcon::~KpkUpdateIcon()
 
 void KpkUpdateIcon::showSettings()
 {
-    QProcess::execute("kpackagekit", QStringList() << "--settings");
+    KToolInvocation::kdeinitExec("kpackagekit", QStringList() << "--settings");
 }
 
 // refresh the cache and try to update,
@@ -265,7 +266,7 @@ void KpkUpdateIcon::autoUpdatesFinished(
 
 void KpkUpdateIcon::showUpdates()
 {
-    QProcess::execute("kpackagekit", QStringList() << "--updates");
+    KToolInvocation::kdeinitExec("kpackagekit", QStringList() << "--updates");
 }
 
 void KpkUpdateIcon::removeStatusNotifierItem()
